src/storage.h

```diff
@@ -0,0 +1,226 @@
+
+  /// @brief the in commodity
+  inline void in_commods_(std::string c) { in_commods.push_back(c); }
+  inline std::vector<std::string> in_commods_() const { return in_commods; }
+
+  /// @brief the in commodity preferences
+  inline void in_commod_prefs_(double c) { in_commod_prefs.push_back(c); }
+  inline double in_commod_prefs_() const { return in_commod_prefs.front(); }
+
+  /// @brief the out commodity
+  inline void out_commods_(std::string c) { out_commods.push_back(c); }
+  inline std::string out_commods_() const {
+    if (out_commods.size() == 0){
+      return "out_commod";
+    }
+     return out_commods.front(); }
+
+  /// @brief the in recipe
+  inline void in_recipe_(std::string c) { in_recipe = c; }
+  inline std::string in_recipe_() const { return in_recipe; }
+
+  /// @brief current maximum amount that can be added to processing
+  inline double current_capacity() const { 
+    return (max_inv_size - processing.quantity() - stocks.quantity()); }
+
+  /// @brief the batch handling identifier
+  inline void batch_handling_(bool c) { batch_handling = c; }
+  inline bool batch_handling_() const { return batch_handling; }
+
+  /// @brief returns the time key for ready materials
+  int ready_time(){ return context()->time() - residence_time; }
+
+ protected:
+  ///   @brief adds a material into the incoming commodity inventory
+  ///   @param mat the material to add to the incoming inventory.
+  ///   @throws if there is trouble with pushing to the inventory buffer.
+  void AddMat_(cyclus::Material::Ptr mat);
+
+  /// @brief Move all unprocessed inventory to processing
+  void BeginProcessing_();
+
+  /// @brief Move as many ready resources as allowable into stocks
+  /// @param cap current throughput capacity 
+  void ProcessMat_(double cap);
+
+  /// @brief any ready resources remaining in processing get pushed off to next timestep
+  /// @param time the timestep whose buffer remains unprocessed 
+  // void AdvanceUnconverted_(int time);
+
+  /// @brief move ready resources from processing to ready at a certain time
+  /// @param time the time of interest
+  void ReadyMatl_(int time);
+
+  /* --- Module Members --- */
+
+  #pragma cyclus var {"tooltip":"input commodity",\
+                      "doc":"commodity accepted by this facility"}
+  std::vector<std::string> in_commods;
+
+  #pragma cyclus var {"default": [1]}
+  std::vector<double> in_commod_prefs;
+
+  #pragma cyclus var {"tooltip":"output commodity",\
+                      "doc":"commodity produced by this facility"}
+  std::vector<std::string> out_commods;
+
+  #pragma cyclus var {"default":"",\
+                      "tooltip":"input recipe",\
+                      "doc":"recipe accepted by this facility"}
+  std::string in_recipe;
+
+  #pragma cyclus var {"default": 0,\
+                      "tooltip":"residence time (timesteps)",\
+                      "doc":"the minimum holding time for a received commodity (timesteps)."}
```

Comment:
Do we really want this to have a default? - I'd lean towards not having a default - since this is the primary reason someone would use this facility.